test/parallel/test-http-header-overflow.js

```diff
@@ -21,7 +21,8 @@ server.on('connection', mustCall((socket) => {
     type: Error,
     message: 'Parse Error',
     code: 'HPE_HEADER_OVERFLOW',
-    bytesParsed: maxHeaderSize + PAYLOAD_GET.length,
+    // -1 is workaround for HTTPParser issue
+    bytesParsed: maxHeaderSize + PAYLOAD_GET.length - 1,
```

Comment:
Isn't the `-1` actually correct? We **shouldn't** parse the whole payload due to the overflow? @mcollina?

Reply:
@indutny could you take a look at this discrepancy as well?

Reply:
File with llhttp with -1

Reply:
There is a difference between how the limits applied in both parsers.

I'll take a look at this, but have a feeling that this behavior might be inherent to the principles of `llhttp`. Note that it still applies the limit properly, the maximum amount of the data allocated to store headers by the both parsers before hitting the limit should be quivalent.

Reply:
I think the easy fix would be to change `Cookie: ` to `Cookie:` above. The space is not accounted for in `llhttp`. Could you give it a try, please?

Reply:
@indutny: still same problem https://travis-ci.com/nodejs/node/jobs/206615243#L272

Reply:
I'm actually unsure if the bug is in legacy of llhttp. Only thing I do know is that they differ.

Reply:
Ah, now I see what's going on. It is not ` ` in the header, it is the colon between header name and value that does not count towards the limit. It seems that there is no way to fix it properly, and I'm not sure if the fix would be of any value outside of the test.

What do you think?

Reply:
Not sure I have much opinion in the matter. What should we do? @lpinca

Reply:
I don’t think we should fix this discrepancy, but just make the test take into account if it’s running on llhttp or not.